src/DI/TranslationExtension.php

```diff
@@ -142,7 +142,7 @@ public function loadConfiguration()
 			->setClass(MessageSelector::class);
 
 		$builder->addDefinition($this->prefix('formatter'))
-			->setClass(MessageFormatter::class);
+			->setClass(MessageFormatter::class, [$this->prefix('@selector'), NULL]);
```

Comment:
Are you sure this is the correct fix? I'm not sure what is Translator used for in MessageFormatter and if it should be the main translator or not.

Reply:
Well, I'm not 100% sure if this is the correct fix, but according to the changes in the [MessageFormatter constructor](https://github.com/symfony/translation/compare/v4.1.8...v4.2.0#diff-109796ab2acce747355124fc98cac6bcL26) in symfony, only the second argument (which causes the circular reference issue) is new, the first one should be the same as before (the first `if` in the constructor is for backward compatibility, I think).

Reply:
The second parameter is IntlFormatterInterface, right? How can that cause any problem when the [implementation](https://github.com/symfony/translation/blob/master/Formatter/IntlFormatter.php) has no dependencies?

Reply:
Oh I see, MessageFormatter works as a proxy for IntlFormatterInterface.

Can you add IntlFormatter as a new service?

Reply:
Sure, I commited the changes